next-app/src/routes/projects/[project_code]/meta/types.d.ts

```diff
@@ -0,0 +1,22 @@
+type ProjectDetails = {
+	id: string,
+	code: string,
+	name: string,
+	num_users: number,
+	num_entries: number,
+	num_entries_with_audio: number,
+	num_entries_with_pictures: number,
+	num_unresolved_comments?: number,
+}
+
+type Entry = {
+	senses?: Sense[],
+	[key: string]: Field,
+}
+
+interface Sense extends Field {
+	examples?: Field[],
+	pictures?: Field[],
+}
+
+type Field = [string, string | Field | Field[]]
```

Comment:
This one has me really confused. This type would match something like this: `['some string value', 'string, or another field or list of fields']` meaing that a field is always an array, but I don't think that's correct, taking a look at the json there's no array where you're using field. Instead I would define a field like this:
```ts
interface Field {
  value: string;
}
```

doing that and fixing how entry is structured might help cleanup some type errors.

Reply:
as above, let's review the structure of the data and see what is the right way to represent the types.